src/main/java/com/github/complate/nashorn/renderer/NashornComplateRenderer.java

```diff
@@ -37,10 +45,29 @@
             "    var console = { log: print, error: print };\n" +
             "}\n\n";
 
-    private final NashornScriptEngine engine;
+    private final ThreadLocal<NashornScriptEngine> engine = new ThreadLocal<NashornScriptEngine>() {
+        @Override
+        protected NashornScriptEngine initialValue() {
+            return initEngine();
+        }
+    };
+
+    private final ComplateScript script;
+    private final Map<String, ?> bindings;
 
     private NashornComplateRenderer(ComplateScript script, Map<String, ?> bindings) {
-        engine = createEngine(bindings);
+        this.script = script;
+        this.bindings = bindings;
+    }
+
+    private NashornScriptEngine initEngine() {
+        final NashornScriptEngine engine = (NashornScriptEngine) new NashornScriptEngineFactory().getScriptEngine();
```

Comment:
I'm not 100% sure but in case `new NashornScriptEngineFactory().getScriptEngine()` returns `null` (which is checked for the line below) using the cast here would not work.